File: internal/templates/templates/todo-api/_python/services/create-todo/handler.py
"""POST /todos — add a todo."""

import json
import os
import uuid

import boto3

table = boto3.resource("dynamodb").Table(os.environ["TABLE_NAME"])


def handler(event, context):
    body = json.loads(event.get("body") or "{}")
    if not body.get("text"):
        return {"statusCode": 422, "body": json.dumps({"error": "text is required"})}

    todo = {"id": str(uuid.uuid4()), "text": body["text"], "done": False}
    table.put_item(Item=todo)

    print(f"created todo {todo['id']}: {todo['text']}")
    return {"statusCode": 201, "body": json.dumps(todo)}

File: internal/templates/templates/todo-api/_node/services/create-todo/handler.mjs
// POST /todos — add a todo.
import { randomUUID } from "node:crypto";
import { DynamoDBClient } from "@aws-sdk/client-dynamodb";
import { DynamoDBDocumentClient, PutCommand } from "@aws-sdk/lib-dynamodb";

const db = DynamoDBDocumentClient.from(new DynamoDBClient({}));

export const handler = async (event) => {
  const body = JSON.parse(event.body ?? "{}");
  if (!body.text) {
    return { statusCode: 422, body: JSON.stringify({ error: "text is required" }) };
  }

  const todo = { id: randomUUID(), text: body.text, done: false };
  await db.send(new PutCommand({ TableName: process.env.TABLE_NAME, Item: todo }));

  console.log(`created todo ${todo.id}: ${todo.text}`);
  return { statusCode: 201, body: JSON.stringify(todo) };
};
